src/world/world.cpp

```diff
@@ -0,0 +1,89 @@
+#include "world.h"
+
+World::World() {
+
+}
+
+void World::addEntity(Entity *entity, int entityPriotity) {
+    // Check if priority is already created
+    if(!_worldEntities.keys().contains(entityPriotity)) {
+        // Create it
+        _worldEntities.insert(entityPriotity, new QHash<int, Entity*>());
+    }
+
+    // Take entities registered with 'entityPriority' priority
+    QHash<int, Entity*> *prioEntities = _worldEntities.value(entityPriotity);
+
+    // Inser them in hash using their id (unique in the software) as key
+    prioEntities->insert(entity->entityId(), entity);
+
+    // Set entity priority
+    entity->setPriority(entityPriotity);
+}
+
+void World::startEntities() {
+    // Get priorities in hash
+    QList<int> priorities = _worldEntities.keys();
+
+    // In each priority (decreasing, most priority first)
+    const int prioritiesSize = priorities.size();
+    for(int i = prioritiesSize - 1; i >= 0; i--) {
```

Comment:
This will take priorities sorted by the higher to lower, **so the entity with the most higher priority will be started first!**